feat: add diff suppression to `okta_network_zone`

exitcode0 opened the pull request:
## Context:
When provided an individual IP address, the Okta Network zone API normalizes to an IP range 
e.g. `1.2.3.4` => `1.2.3.4-1.2.3.4`

## Proposed Changes:
This PR adds diff suppression to ensure that IP addresses normalized by the Okta API do not cause a diff in subsequent `Terraform Plan` outputs

## Possible Risks:
While (IMO) unlikely, The Okta API may one day change how it approaches normalizing individual IP Addresses
- The API may begin rejecting individual IP addresses
- The API may change how it normalizes IPs over to normalizing to a `/32` CIDR address
  - e.g. `1.2.3.4` => `1.2.3.4/32`
  
fixes #2471

exitcode0 committed:
add diffSuppression to okta_network_zone

exitcode0 committed:
feat: migrate network zone test to new framework

- Update TestAccResourceOktaNetworkZone_ipNormalization to use new test
  framework
- Replace deprecated test functions with acctest equivalents:
  - oktaResourceTest → acctest.OktaResourceTest
  - testAccPreCheck → acctest.AccPreCheck
  - testAccProvidersFactories → acctest.ProtoV5ProviderFactoriesForTestAcc
  - buildResourceName → acctest.BuildResourceName
- Use resources.OktaIDaaSNetworkZone constant instead of networkZone
  variable
- Add VCR test fixture for IP normalization test to improve test
  reliability

exitcode0 committed:
fix: re-record VCR cassettes

The cassette was originally recorded against the v5 SDK which did not
make separate lifecycle/activate API calls after zone creation. The v6
SDK upgrade added explicit activate/deactivate calls, causing VCR
playback to fail with "requested interaction not found" errors in CI.

pranav-okta reviewed (changes requested):
Oops, I forgot to publish my comments. PTAL.

exitcode0 commented on okta/services/idaas/resource_okta_network_zone.go:
```diff
@@ -70,6 +81,7 @@ func resourceNetworkZone() *schema.Resource {
 			"status": {
 				Type:         schema.TypeString,
 				Optional:     true,
+				Default:      "ACTIVE",
```
defaulting to `"ACTIVE"` matches the behavior I saw in testing when the `status` attribute is omitted from `POST /api/v1/zones`

exitcode0 committed:
chore: address PR review feedback

- Remove unused ACC_TEST_BASE variable from Makefile
- Remove stale "add support for IP types" TODO comments from gateways/proxies schema attributes in resource_okta_network_zone.go

exitcode0 commented on okta/services/idaas/resource_okta_network_zone.go:
```diff
@@ -49,7 +50,12 @@ func resourceNetworkZone() *schema.Resource {
 				Optional:    true,
 				Description: "Array of values in CIDR/range form depending on the way it's been declared (i.e. CIDR will contain /suffix). Please check API docs for examples. Use with type `IP`",
-				Elem:        &schema.Schema{Type: schema.TypeString},
+				// TODO: Next Major Release - add support for IP types
```
done

exitcode0 commented on Makefile:
```diff
@@ -75,6 +75,9 @@ test:
 	echo $(UNIT_TESTS) | \
 		xargs -t -n4 go test $(TESTARGS) $(TEST_FILTER) -timeout=30s -parallel=4
 
+# Base acceptance test configuration
```
done
